feat: Add AllowHtml attribute for entity properties

raffaelecarelle opened the pull request:
### 1. Why is this change necessary?
Add the possibility from Attribute entity allow html in text fields

### 2. What does this change do, exactly?
Add Flag AllowHtml from entity.

raffaelecarelle committed:
Add AllowHtml attribute for entity properties

Introduced a new `#[AllowHtml]` attribute to mark entity properties that allow HTML content. This includes support for an optional `sanitized` flag and updates to the AttributeEntityCompiler to process the new attribute. A changelog entry has also been added for visibility.

raffaelecarelle committed:
fix cs

mitelg commented on changelog/_unreleased/2025-01-02-add-allow-html-entity-attribute.md:
```diff
@@ -0,0 +1,8 @@
+---
+author: Raffaele Carelle
+title: Add `#[AllowHtml]` entity attribute
+issue: NEXT-00000
```
can you change the changelog similar to your other PR? https://github.com/shopware/shopware/pull/6637

raffaelecarelle replied:
Fixed

mitelg commented on src/Core/Framework/DataAbstractionLayer/Attribute/AllowHtml.php:
```diff
@@ -0,0 +1,14 @@
+
+use Shopware\Core\Framework\Log\Package;
+
+#[Package('core')]
```
```suggestion
#[Package('framework')]
```

raffaelecarelle replied:
Fixed

mitelg reviewed (changes requested):
would you mind adding tests like in your other PR? https://github.com/shopware/shopware/pull/6637

raffaelecarelle committed:
Add AllowHtml Attribute support

Updated the changelog entry with corrected title and metadata. Clarifications include the author's email, GitHub handle, and a refined description of the AllowHtml attribute functionality.

raffaelecarelle committed:
Update Package annotation in AllowHtml attribute

Changed the Package annotation from 'core' to 'framework' to better reflect its contextual grouping. This helps maintain consistency and clarity in the codebase.

raffaelecarelle commented:
> would you mind adding tests like in your other PR? #6637

Hi @mitelg test added

mitelg commented on Makefile:
please remove again :grin:

raffaelecarelle replied:
ops :D removed

mitelg commented:
@raffaelecarelle nice! thank you :+1: just one thing and we are good to go

raffaelecarelle committed:
Add htmlString field to AttributeEntity and test data

The `htmlString` field has been added with `AllowHtml` support to the `AttributeEntity` and incorporated into multiple test cases. This ensures proper handling, validation, and coverage of HTML-containing strings throughout the test scenarios. Additionally, corresponding schema changes have been updated to reflect this new field.

mitelg commented:
thanks for your contribution @raffaelecarelle 👍🎉💙
